comp/metadata/inventorysoftware/impl/inventorysoftware.go

```diff
@@ -0,0 +1,194 @@
+	sysProbeClient SysProbeClient
+	// cachedInventory stores the most recently collected software inventory data
+	cachedInventory []software.Entry
+	// hostname identifies the system where the inventory was collected
+	hostname string
+	// enabled indicates whether software inventory collection is enabled in the configuration
+	enabled bool
+}
+
+// Requires defines the dependencies required by the inventory software component.
+// This struct defines all the required dependencies that must be provided
+// when creating a new inventory software component instance.
+type Requires struct {
+	// Log provides logging capabilities for the component
+	Log log.Component
+	// Config provides access to the agent configuration
+	Config config.Component
+	// Serializer is used to serialize and send data to the backend
+	Serializer serializer.MetricSerializer
+	// Hostname provides the hostname of the current system
+	Hostname hostnameinterface.Component
+}
+
+// Provides defines the output of the inventory software component.
+// This struct defines all the services and providers that the component
+// makes available to the rest of the system.
+type Provides struct {
+	// Comp is the main component interface for software inventory
+	Comp inventorysoftware.Component
+	// Provider is the metadata provider for software inventory data
+	Provider runnerimpl.Provider
+	// FlareProvider provides software inventory data for flare collection
+	FlareProvider flaretypes.Provider
+	// StatusHeaderProvider provides status information for the agent status page
+	StatusHeaderProvider status.HeaderInformationProvider
+	// Endpoint provides HTTP endpoint access to software inventory data
+	Endpoint api.AgentEndpointProvider
+}
+
+// New creates a new inventory software component with the default sysprobeclient
+func New(reqs Requires) (Provides, error) {
+	return NewWithClient(reqs, nil)
+}
+
+// NewWithClient creates a new inventory software component with a custom sysprobeclient
+func NewWithClient(reqs Requires, client SysProbeClient) (Provides, error) {
+	if client == nil {
+		client = &sysProbeClientWrapper{
+			client: sysprobeclient.GetCheckClient(pkgconfigsetup.SystemProbe().GetString("system_probe_config.sysprobe_socket")),
+		}
+	}
+	hname, _ := reqs.Hostname.Get(context.Background())
+	is := &inventorySoftware{
+		log:            reqs.Log,
+		sysProbeClient: client,
+		hostname:       hname,
+		enabled:        reqs.Config.GetBool("software_inventory.enabled"),
+	}
+	// Always provide the component because FX dependency injection expects
+	// status providers, metadata providers, etc. to be available, not wrapped
+	// in option.Option. The enabled flag only controls whether we call the System
+	// Probe software inventory endpoint to collect actual data.
+	// Note that there is a second way to disable this feature, through InventoryPayload.Enabled.
+	// 'enable_metadata_collection' and 'inventories_enabled' both need to be set to true.
```

Comment:
If the software inventory is not enabled, we can return `nil` flare providers, status provider, etc. The components that process those filter out nil providers. If you are curious, you can search for `fxutil.GetAndFilterGroup`

That way if is disable it won't show in the status output, flare, etc 😄